src/app/view/components/dot-device-selector/dot-device-selector.component.ts

```diff
@@ -14,22 +17,28 @@ export class DotDeviceSelectorComponent implements OnInit {
     @Input() value: DotDevice;
     @Output() selected = new EventEmitter<DotDevice>();
 
-    options: Observable<DotDevice[]>;
+    options: DotDevice[];
+    readonly arrowDropdownComponentSize = 32;
+    dropdownWidth: number;
 
-    constructor(private dotDevicesService: DotDevicesService, private dotMessageService: DotMessageService) {}
+    constructor(
+        private dotDevicesService: DotDevicesService,
+        private dotMessageService: DotMessageService
+    ) {}
 
     ngOnInit() {
-        this.options = this.dotMessageService
-            .getMessages(['editpage.viewas.default.device'])
-            .mergeMap(() =>
-                this.dotDevicesService
-                    .get()
-                    .pipe(
-                        map((devices: DotDevice[]) =>
-                            this.setOptions(this.dotMessageService.get('editpage.viewas.default.device'), devices)
-                        )
-                    )
-            );
+        this.dotMessageService.getMessages(['editpage.viewas.default.device']).subscribe(() => {
```

Comment:
You don't need to get rid of the observable here, you can do a `tap` just for the `setDropdownWidth` and everything else remains the same.

Reply:
Ok, not we don't have auto unsubscribe, we need a `take(1)` in the `getMessages` too.